server/app/views/applicant/ApplicantProgramSummaryView.java

```diff
@@ -175,49 +163,44 @@ private DivTag renderQuestionSummary(
       actionAndTimestampDiv.with(timestampContent);
     }
 
-    // Maybe link to block containing specific question.
-    if (data.isAnswered() || isFirstUnanswered) {
-      String editText = messages.at(MessageKey.LINK_EDIT.getKeyName());
-      String ariaLabel = messages.at(MessageKey.ARIA_LABEL_EDIT.getKeyName(), data.questionText());
-      if (!data.isAnswered()) {
-        if (inReview) {
-          editText = messages.at(MessageKey.BUTTON_CONTINUE.getKeyName());
-          ariaLabel = messages.at(MessageKey.ARIA_LABEL_CONTINUE.getKeyName(), data.questionText());
-        } else {
-          editText = messages.at(MessageKey.LINK_BEGIN.getKeyName());
-          ariaLabel = messages.at(MessageKey.ARIA_LABEL_BEGIN.getKeyName(), data.questionText());
-        }
-      }
-      String editLink =
-          (!data.isAnswered() && !inReview)
-              ? routes.ApplicantProgramBlocksController.edit(
+    LinkElement editElement =
+        new LinkElement()
+            .setStyles("bottom-0", "right-0", "text-blue-600", StyleUtils.hover("text-blue-700"));
+    if (data.isAnswered()) {
+      editElement
+          .setHref(
+              routes.ApplicantProgramBlocksController.review(
                       applicantId, data.programId(), data.blockId())
-                  .url()
-              : routes.ApplicantProgramBlocksController.review(
+                  .url())
+          .setText(messages.at(MessageKey.LINK_EDIT.getKeyName()))
+          .setIcon(Icons.EDIT)
+          .setIconPosition(LinkElement.IconPosition.START);
+    } else {
+      editElement
+          .setHref(
+              routes.ApplicantProgramBlocksController.edit(
```

Comment:
Is there a difference between ApplicantProgramBlocksController.edit and ApplicantProgramBlocksController.review? I just tried on staging and html looked the same.

Reply:
So glad you asked... The difference is what is the next page it will go when the user clicks "save and next"
https://github.com/civiform/civiform/blob/b1157e5bea524ec28db1dec4450e2c1013b3cf6a/server/app/controllers/applicant/ApplicantProgramBlocksController.java#L92-L121

I've looked at that code, and I'm not convinced it actually makes sense to have that distinction, and some of the transitions seem confusing there (e.g. it could navigate backwards). 
It's also not very well tested.
I even started to simplify that logic https://github.com/civiform/civiform/pull/3823. Definitely out of scope for this PR, happy to talk separately.